src/main/java/org/apache/commons/compress/archivers/sevenz/HeaderChannelBuffer.java

```diff
@@ -0,0 +1,174 @@
+ * the buffer will be fully read and held in a {@link HeaderInMemoryBuffer}.
+ *
+ * @NotThreadSafe
+ * @since 1.21
+ */
+class HeaderChannelBuffer implements HeaderBuffer {
+    private static final int DEFAULT_PAGE_MAX = 16_777_216;
+    // This must match the largest get<Element> (currently getLong)
+    private static final int MAX_GET_ELEMENT_SIZE = 8;
+    private final ReadableByteChannel channel;
+    private final ByteBuffer buffer;
+    private long remaining;
+
+    private HeaderChannelBuffer(final ReadableByteChannel channel, final long capacity, final int maxPageBytes) {
+        this.channel = channel;
+        int limit = (int) Math.min(maxPageBytes, capacity);
+        this.buffer = ByteBuffer.allocate(limit).order(ByteOrder.LITTLE_ENDIAN);
+        this.remaining = capacity;
+    }
+
+    public static HeaderBuffer create(final ReadableByteChannel channel, final long capacity, final int maxPageBytes)
+            throws IOException {
+        if (maxPageBytes < MAX_GET_ELEMENT_SIZE) {
+            throw new IllegalArgumentException("Page size must be at least " + MAX_GET_ELEMENT_SIZE);
+        }
+        if (capacity <= maxPageBytes) {
+            ByteBuffer buf = ByteBuffer.allocate((int) capacity).order(ByteOrder.LITTLE_ENDIAN);
+            IOUtils.readFully(channel, buf);
+            buf.flip();
+            return new HeaderInMemoryBuffer(buf);
+        }
+        HeaderChannelBuffer channelBuffer = new HeaderChannelBuffer(channel, capacity, maxPageBytes);
+        channelBuffer.fill();
+        return channelBuffer;
+    }
+
+    public static HeaderBuffer create(final ReadableByteChannel channel, final long capacity) throws IOException {
+        return HeaderChannelBuffer.create(channel, capacity, DEFAULT_PAGE_MAX);
+    }
+
+    public static HeaderBuffer create(final InputStream inputStream, final long capacity, final int maxPageBytes)
+            throws IOException {
+        return create(Channels.newChannel(inputStream), capacity, maxPageBytes);
+    }
+
+    public static HeaderBuffer create(final InputStream inputStream, final long capacity) throws IOException {
+        return create(Channels.newChannel(inputStream), capacity, DEFAULT_PAGE_MAX);
+    }
+
+    @Override
+    public boolean hasCRC() {
+        return false;
+    }
+
+    @Override
+    public CRC32 getCRC() throws IOException {
+        throw new IOException("CRC is not implemented for this header type");
+    }
+
+    @Override
+    public void get(byte[] dst) throws IOException {
+        int remainingBytes = dst.length;
+        do {
+            int length = Math.min(buffer.remaining(), remainingBytes);
+            buffer.get(dst, dst.length - remainingBytes, length);
+            remainingBytes -= length;
+        } while (refilled(remainingBytes));
+    }
+
+    private boolean refilled(final int remainingBytes) throws IOException {
+        if (remainingBytes <= 0) {
+            return false;
+		}
```

Comment:
Fix formatting please.

Reply:
Hi, do we think this PR is still appropriate? I currently have no files large enough to test the issue it resolved, and haven't found time to create one yet. No problem with closing it, but if it's still useful I'll try to make time for it again.